demo/c/test_demo.py

```diff
@@ -0,0 +1,143 @@
+# file accompanying this source.
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+
+import logging
+import os
+import platform
+import subprocess
+import sys
+import unittest
+
+from pvporcupine import util
+
+class PorcupineDemoTestCase(unittest.TestCase):
+    @staticmethod
+    def __has_detected(s):
+        return "detected" in s
+
+    @staticmethod
+    def __append_language(s, language):
+        if language == 'en':
+            return s
+        return f'{s}_{language}'
+
+    @classmethod
+    def setUpClass(cls):
+        cls.__ENVIRONMENT = cls.__get_environemnt()
+
+    def setUp(self):
+        self.command = ["./build/porcupine_demo_file",
+                    "-l", util.pv_library_path('.'),
+                    "-m", util.pv_model_path('.'),
+                    "-k", "",
+                    "-t", "0.5",
+                    "-w", "",
+                    "-a", sys.argv[1]]
+
+    def test_americano(self):
+        self.command[6] = self.__pv_keyword_path("en", "americano")
+        self.command[10] = self.__pv_audio_path("multiple_keywords.wav")
+
+        run_demo = subprocess.run(self.command, capture_output=True, text=True)
+
+        self.assertEqual(run_demo.returncode, 0)
+        self.assertTrue(self.__has_detected(run_demo.stdout))
+
+    def test_heuschrecke_de(self):
+        language = "de"
+        self.command[4] = self.__pv_model_path(language)
+        self.command[6] = self.__pv_keyword_path(language, "heuschrecke")
+        self.command[10] = self.__pv_audio_path("heuschrecke.wav")
+
+        run_demo = subprocess.run(self.command, capture_output=True, text=True)
+
+        self.assertEqual(run_demo.returncode, 0)
+        self.assertTrue(self.__has_detected(run_demo.stdout))
+
+    def test_manzana_es(self):
+        language = "es"
+        self.command[4] = self.__pv_model_path(language)
+        self.command[6] = self.__pv_keyword_path(language, "manzana")
+        self.command[10] = self.__pv_audio_path("manzana.wav")
+
+        run_demo = subprocess.run(self.command, capture_output=True, text=True)
+
+        self.assertEqual(run_demo.returncode, 0)
+        self.assertTrue(self.__has_detected(run_demo.stdout))
+
+    def test_mon_chouchou_fr(self):
+        language = "fr"
+        self.command[4] = self.__pv_model_path(language)
+        self.command[6] = self.__pv_keyword_path(language, "mon chouchou")
+        self.command[10] = self.__pv_audio_path("mon_chouchou.wav")
+
+        run_demo = subprocess.run(self.command, capture_output=True, text=True)
+
+        self.assertEqual(run_demo.returncode, 0)
+        self.assertTrue(self.__has_detected(run_demo.stdout))
+
+    @staticmethod
+    def __get_environemnt():
```

Comment:
I wish there was a way to use only functions from `utils` for stuff like this, but I see now that there isn't any public function available to handle this particular need. We can probably figure out a way to do this in the future updates of the binding so that we don't need this function at all.